lxd/instance/drivers/driver_lxc.go

```diff
@@ -2564,6 +2564,19 @@ func (d *lxc) onStart(_ map[string]string) error {
 	if d.expandedConfig["limits.network.priority"] != "" {
 		go func(d *lxc) {
 			d.fromHook = false
+
+			timeout := time.Now().Add(10 * time.Second)
+
+			for !d.IsRunning() {
```

Comment:
Why is this needed?

Reply:
container should be fully started at the point where we try to configure network priority,
because in `setNetworkPriorityWithCgroup` we are accessing the cgroup stuff and LXC API.

Reply:
mmm yeah, but was this not working before?

Reply:
yes. It was not working. I was surprised too.

Reply:
You can easily reproduce that by making container stop (with network limit set) and then start the container. Limit won't be set.

Reply:
Hrm ok, let me think about this for a bit as it seems like maybe we should trigger this event later rather than polling for it the instance being ready.

Reply:
> Is this something we can add a test for in the test suite?

theoretically yes, to check that we need to dump firewall rules and parse them OR (in cgroup case) find cgroup and check netprio cgroup file content.